From ad81de2c1d4be2c13df7689e5bbbe0feb204d353 Mon Sep 17 00:00:00 2001
Message-Id: <ad81de2c1d4be2c13df7689e5bbbe0feb204d353.1650636481.git.keller@fs-net.de>
In-Reply-To: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
References: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
From: Li Feng <lifeng@codeaurora.org>
Date: Thu, 26 Sep 2019 16:13:10 +0800
Subject: [PATCH 08/56] qcacld-2.0: Allocate memory for chain stats by default

Allocate memory for chain stats by default in case of oob.

Change-Id: I42a2cc5b4ab2df09e33b87a8a5a185cbdd61476c
CRs-Fixed: 2511085
---
 CORE/SERVICES/WMA/wma.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CORE/SERVICES/WMA/wma.c b/CORE/SERVICES/WMA/wma.c
index 22e0cfa..aeb93a1 100644
--- a/CORE/SERVICES/WMA/wma.c
+++ b/CORE/SERVICES/WMA/wma.c
@@ -3285,15 +3285,15 @@ static int wma_stats_event_handler(void *handle, u_int8_t *cmd_param_info,
 	buf_data_size = buf_size - sizeof(*event);
 
 	rssi_event = param_buf->chain_stats;
+	buf_size += sizeof(*rssi_event);
 	if (rssi_event) {
 		if ((((rssi_event->tlv_header & 0xFFFF0000) >> 16) ==
 				WMITLV_TAG_STRUC_wmi_per_chain_rssi_stats) &&
 			 ((rssi_event->tlv_header & 0x0000FFFF) ==
 				WMITLV_GET_STRUCT_TLVLEN(
 					wmi_per_chain_rssi_stats))) {
-			buf_size += sizeof(*rssi_event) +
-				(rssi_event->num_per_chain_rssi_stats *
-				sizeof(wmi_rssi_stats));
+			buf_size += rssi_event->num_per_chain_rssi_stats *
+				sizeof(wmi_rssi_stats);
 			rssi_stats_support = TRUE;
 		}
 	}
-- 
2.14.5

